sqlmesh/core/config/model.py

```diff
@@ -45,6 +46,7 @@ class ModelDefaultsConfig(BaseConfig):
     on_destructive_change: t.Optional[OnDestructiveChange] = None
     session_properties: t.Optional[t.Dict[str, t.Any]] = None
     audits: t.Optional[t.List[FunctionCall]] = None
+    optimize: t.Optional[bool] = None
```

Comment:
Isn't this only applicable to SQL model? Should we make sure this is not set for models of other kinds?

Reply:
Yep, it does make sense for it to be exclusive to `SQLModel`. Moving it there would add some (small) complexity such as the need to override `data_hash_values` afaict; Is this fine or would it be better to leave it and instead emit a warning/error?

Reply:
Should be fine, we already do this for other attributes exclusive to it.

Reply:
Ended up leaving it in `ModelMeta` and added a validation check